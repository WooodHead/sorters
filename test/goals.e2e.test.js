import cheerio from 'cheerio'
import pretty from 'pretty'
import {ObjectID} from 'mongodb'
import {setup, teardown} from './setup'
import {generateAndLogUser, setUserData} from './fixtures'

let db
let browser

beforeAll(async () => {
    const res = await setup()
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()
})

afterAll(async () => {
    await teardown({db, browser})
})

describe('goals', () => {
    it('displays', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)
        await setUserData(browserPage)
        
        const status = await browserPage.open(`http://localhost:3000/account/goals`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const html = pretty(page.html())
        expect(html).toMatchSnapshot()
    })
})