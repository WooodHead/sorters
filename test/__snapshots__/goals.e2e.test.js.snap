// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goals displays 1`] = `
"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-1824255950\\">
  <div data-reactid=\\"2\\">
    <!-- react-empty: 3 -->
    <div style=\\"padding-top:50px;padding-bottom:20px;\\" data-reactid=\\"4\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\" data-reactid=\\"5\\">
        <div class=\\"container\\" data-reactid=\\"6\\">
          <div class=\\"navbar-header\\" data-reactid=\\"7\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\" data-reactid=\\"8\\"><span class=\\"sr-only\\" data-reactid=\\"9\\">Toggle navigation</span><span class=\\"icon-bar\\" data-reactid=\\"10\\"></span><span class=\\"icon-bar\\" data-reactid=\\"11\\"></span><span class=\\"icon-bar\\" data-reactid=\\"12\\"></span></button>
            <a class=\\"navbar-brand\\"
              href=\\"/\\" data-reactid=\\"13\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\" data-reactid=\\"14\\">
            <ul class=\\"nav navbar-nav navbar-right\\" data-reactid=\\"15\\">
              <li class=\\"false\\" data-reactid=\\"16\\">
                <a href=\\"/\\" data-reactid=\\"17\\">Home</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"18\\">
                <a href=\\"/about\\" data-reactid=\\"19\\">About</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"20\\">
                <a href=\\"/news\\" data-reactid=\\"21\\">News</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"22\\">
                <a href=\\"/users\\" data-reactid=\\"23\\">Sorters</a>
              </li>
              <li class=\\"false\\" data-reactid=\\"24\\">
                <a href=\\"/account\\" data-reactid=\\"25\\">Dashboard</a>
              </li>
              <li class=\\"dropdownfalse\\" data-reactid=\\"26\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\" data-reactid=\\"27\\">
                  <!-- react-text: 28 -->Hello,
                  <!-- /react-text -->
                  <!-- react-text: 29 -->test
                  <!-- /react-text -->
                  <!-- react-text: 30 -->
                  <!-- /react-text --><span class=\\"caret\\" data-reactid=\\"31\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div style=\\"max-width:400px;margin:auto;\\" data-reactid=\\"32\\">
      <h1 data-reactid=\\"33\\">Goals</h1>
      <div data-reactid=\\"34\\">
        <p data-reactid=\\"35\\">
          <!-- react-text: 36 -->Your public goal list can be found at
          <!-- /react-text -->
          <a href=\\"/u/test/goals\\" data-reactid=\\"37\\">
            <!-- react-text: 38 -->/u/
            <!-- /react-text -->
            <!-- react-text: 39 -->test
            <!-- /react-text -->
            <!-- react-text: 40 -->/goals
            <!-- /react-text -->
          </a>
          <!-- react-text: 41 -->.
          <!-- /react-text -->
        </p>
        <h2 data-reactid=\\"42\\">Description</h2>
        <p data-reactid=\\"43\\">Here you can describe how you chose your goals.</p>
        <form data-reactid=\\"44\\">
          <div class=\\"form-group\\" data-reactid=\\"45\\">
            <label for=\\"goals-description\\" data-reactid=\\"46\\">Description</label>
            <textarea class=\\"form-control\\" rows=\\"4\\" data-reactid=\\"47\\">Test user goals</textarea>
          </div>
          <div class=\\"form-group\\" data-reactid=\\"48\\">
            <button type=\\"submit\\" class=\\"btn btn-primary btn-block\\" data-reactid=\\"49\\">Save</button>
          </div>
        </form>
        <div data-reactid=\\"50\\">
          <h2 data-reactid=\\"51\\">Your Goals</h2>
          <ul data-reactid=\\"52\\">
            <li style=\\"cursor:pointer;clear:both;\\" data-reactid=\\"53\\"><span data-reactid=\\"54\\"><span style=\\"display:block;float:right;\\" data-reactid=\\"55\\"><span data-reactid=\\"56\\"><a style=\\"cursor:pointer;font-size:140%;\\" data-reactid=\\"57\\">&#x1F5D1;</a><!-- react-empty: 58 --></span>
              <a style=\\"cursor:pointer;font-size:140%;\\"
                data-reactid=\\"59\\">&#x270E;</a>
              </span><span data-reactid=\\"60\\">Goal</span></span>
            </li>
            <li style=\\"cursor:pointer;clear:both;\\" data-reactid=\\"61\\"><span data-reactid=\\"62\\"><span style=\\"display:block;float:right;\\" data-reactid=\\"63\\"><span data-reactid=\\"64\\"><a style=\\"cursor:pointer;font-size:140%;\\" data-reactid=\\"65\\">&#x1F5D1;</a><!-- react-empty: 66 --></span>
              <a style=\\"cursor:pointer;font-size:140%;\\"
                data-reactid=\\"67\\">&#x270E;</a>
              </span><span data-reactid=\\"68\\">Goal - doing</span></span>
            </li>
            <li style=\\"cursor:pointer;clear:both;\\" data-reactid=\\"69\\"><span data-reactid=\\"70\\"><span style=\\"display:block;float:right;\\" data-reactid=\\"71\\"><span data-reactid=\\"72\\"><a style=\\"cursor:pointer;font-size:140%;\\" data-reactid=\\"73\\">&#x1F5D1;</a><!-- react-empty: 74 --></span>
              <a style=\\"cursor:pointer;font-size:140%;\\"
                data-reactid=\\"75\\">&#x270E;</a>
              </span><span data-reactid=\\"76\\">Goal - done</span><span data-reactid=\\"77\\"> &#x2714;</span></span>
            </li>
          </ul>
        </div>
        <h3 data-reactid=\\"78\\">New Goal</h3>
        <form data-reactid=\\"79\\">
          <div class=\\"form-group\\" data-reactid=\\"80\\">
            <label for=\\"title\\" data-reactid=\\"81\\">Title</label>
            <input type=\\"text\\" class=\\"form-control\\" placeholder=\\"\\" required data-reactid=\\"82\\">
          </div>
          <div class=\\"form-group\\" data-reactid=\\"83\\">
            <button type=\\"submit\\" class=\\"btn btn-primary btn-block\\" data-reactid=\\"84\\">New Goal</button>
          </div>
        </form>
      </div>
    </div>
    <footer data-reactid=\\"85\\">
      <div class=\\"container\\" data-reactid=\\"86\\">
        <hr data-reactid=\\"87\\">
        <p data-reactid=\\"88\\">
          <!-- react-text: 89 -->Brought to you with &#x2764; by
          <!-- /react-text -->
          <a href=\\"/about\\" data-reactid=\\"90\\">Nick Redmark</a>
          <!-- react-text: 91 -->. Support this platform on
          <!-- /react-text -->
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\" data-reactid=\\"92\\">Patreon</a>
          <!-- react-text: 93 -->.
          <!-- /react-text -->
        </p>
      </div>
    </footer>
  </div>
</div>"
`;
